core_lib/helpers/files.py

```
@@ -13,15 +13,35 @@ def download_file_handle(file: Response, file_handle):
 def download_file(path: str, local_filename: str):
-    file = requests.get(path)
+    # `stream=True` — previously the whole response was buffered into memory
+    # before iter_content() chunked it, OOM-risking large downloads.
+    """
+    Download a file from a remote URL and save it to a local file.
+    
+    Parameters:
+        path (str): The URL of the file to download.
+        local_filename (str): The destination file path.
+    """
+    file = requests.get(path, stream=True)
```

Ruff (0.15.18)

[error] Probable use of `requests` call without timeout

(S113)